client/data/nps/index.js

```diff
@@ -21,8 +21,32 @@ export const updateNps = ( data ) =>
 		} )
 	);
 
-export const updateNpsResponse = ( surveyId, data ) =>
-	withRequestTimeout(
+export const updateNpsResponse = ( surveyId, data ) => {
+	console.log( window.csFormsSetup );
+	if ( window.csFormsSetup || window.csFormsSetup._isWpcom ) {
+		return window
+			.fetch(
+				'https://public-api.wordpress.com/crowdsignal-forms/v1/sites/' +
+					window.csFormsSetup._currentSiteId +
+					'/nps/' +
+					surveyId +
+					'/response?_locale=user',
+				{
+					headers: {
+						accept: 'application/json, */*;q=0.1',
+						'content-type': 'application/json',
+						'x-wp-nonce': window.csFormsSetup._nonce,
+					},
+					referrer: window.csFormsSetup._siteUrl,
+					referrerPolicy: 'strict-origin-when-cross-origin',
+					body: JSON.stringify( data ),
+					method: 'POST',
+					mode: 'cors',
+				}
+			)
+			.then( ( response ) => response.json() );
```

Comment:
there is no error handler, I was hoping we can raise it and handle it on the caller